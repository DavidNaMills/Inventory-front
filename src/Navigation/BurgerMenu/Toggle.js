import React from 'react';

const styleObj = {
    backgroundColor: 'rgb(151, 184, 240)',
    borderRadius: '5px'
}

const Toggle = ({ onClick, isOpen = false }) => {


    const style = isOpen
    ? "hamburger hamburger--collapse is-active"
    : "hamburger hamburger--collapse"
    return (
        <button className={style} type="button" style={styleObj} onClick={onClick}>
            <span className={"hamburger-box"}>
                <span className={"hamburger-inner"}></span>
            </span>
        </button>
    )
}

export default Toggle;