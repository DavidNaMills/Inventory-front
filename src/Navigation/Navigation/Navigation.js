import React from 'react';
import classes from './navigation.module.scss';

import { withOrientationChange } from 'react-device-detect'
import NavItem from './NavItem/NavItem';
import HomeIcon from '@material-ui/icons/Home';
import LocationCityIcon from '@material-ui/icons/LocationCity';
import PeopleIcon from '@material-ui/icons/People';
import LocalCafeIcon from '@material-ui/icons/LocalCafe';
import MenuBookIcon from '@material-ui/icons/MenuBook';
import {useLocale} from '../../Context/LocaleContext';

const Navigation = (props) => {
    const locale = useLocale();
    const { userRoutes, isMobile=false, tempClose=null} = props;
    const width = isMobile ? null : (window.innerWidth-20) / userRoutes.length;

    return (
        <div className={isMobile ? classes.navigationMobile : classes.navigation}>
            {
                userRoutes.map((x, y) => {
                    switch(x.to){
                        case '/dashboard':
                            return <NavItem key={y} to={x.to} label={locale[x.label]} Icon={HomeIcon} width={width} isMobile={isMobile} tempClose={tempClose}/>
                        case '/products':
                            return <NavItem key={y} to={x.to} label={locale[x.label]} Icon={LocalCafeIcon} width={width} isMobile={isMobile} tempClose={tempClose}/>
                        case '/productTypes':
                            return <NavItem key={y} to={x.to} label={locale[x.label]} Icon={MenuBookIcon} width={width} isMobile={isMobile} tempClose={tempClose}/>
                        case '/locations':
                            return <NavItem key={y} to={x.to} label={locale[x.label]} Icon={LocationCityIcon} width={width} isMobile={isMobile} tempClose={tempClose}/>
                        case '/staff':
                            return <NavItem key={y} to={x.to} label={locale[x.label]} Icon={PeopleIcon} width={width} isMobile={isMobile} tempClose={tempClose}/>
                        default:
                            return <NavItem key={y} to={x.to} label={locale[x.label]} width={width} isMobile={isMobile} tempClose={tempClose}/>
                    }
                })
            }
        </div>
    )
};

export default withOrientationChange(Navigation);