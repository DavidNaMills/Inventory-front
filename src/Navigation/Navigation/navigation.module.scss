@import '../../styles/variables/colors.module.scss';

@mixin navigation {
    display: flex;
    background-color: $nav-back;
    border-bottom: solid $nav-border 3px;
    margin-bottom: 15px;
    z-index: 999;
}

.navigation {
    @include navigation;
    justify-content: space-between;
    top: 0;
    position: sticky;
}

.navigationMobile {
    @include navigation;
    position: absolute;
    width: 100%;
    height: 100vh;
    padding-top: 20px;
    padding-bottom: 20px;
    flex-direction: column;
    // justify-content: space-between;
}
