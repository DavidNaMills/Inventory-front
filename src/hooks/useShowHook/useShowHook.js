import {useState} from 'react';

const useShowHook = (initValue=false) =>{
    const [isShow, setShow] = useState(initValue);

    const changeShow = (newValue=false) =>{
        if(typeof(initValue)==='boolean'){
            setShow(newValue);
        } else if (Object.prototype.toString.call(initValue) === '[object Object]'){
            let changed = false;
            const temp = JSON.parse(JSON.stringify(isShow));
            for(let x in newValue){
                if((x in temp)){
                    temp[x] = newValue[x];
                    changed = true;
                }
            };
            if(changed){
                setShow(temp);
            }
        }
    }

    const reverseBoolean = ()=>{
        if(typeof(initValue)==='boolean'){
            setShow(prev=>!prev);
        }
    }

    return{
        isShow,
        changeShow,
        reverseBoolean
    }
}

export default useShowHook;