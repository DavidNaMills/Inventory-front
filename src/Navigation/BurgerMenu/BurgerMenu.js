import React from 'react';
import Toggle from './Toggle';
import Navigation from '../Navigation/Navigation';
import useShowHook from '../../hooks/useShowHook/useShowHook';

const style={
    top: '0',
    position: 'sticky',
    zIndex: '9999'
}

const BurgerMenu = (props) => {
    const {userRoutes, locations} = props;
    const {isShow, reverseBoolean, changeShow} = useShowHook(false);

    return (
        <div style={style}>
            <Toggle onClick={() => reverseBoolean()} isOpen={isShow} />
            {isShow &&
                <Navigation userRoutes={userRoutes} locations={locations} isMobile={true} tempClose={()=>changeShow(false)}/>
            }
        </div>
    )
}

export default BurgerMenu;