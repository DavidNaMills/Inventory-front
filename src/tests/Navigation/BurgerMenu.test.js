import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import BurgerMenu from '../../Navigation/BurgerMenu/BurgerMenu';
import Toggle from '../../Navigation/BurgerMenu/Toggle';
import Navigation from '../../Navigation/Navigation/Navigation';

jest.mock('../../hooks/useShowHook/useShowHook');
import useShowHook from '../../hooks/useShowHook/useShowHook';


describe('BurgerMenu test auite', () => {
    describe('with isShow set to false', () => {
        let wrapper;
        beforeEach(()=>{
            useShowHook.mockImplementation(() => ({
                isShow: false,
                reverseBoolean: jest.fn()
            }));
            wrapper = shallow(<BurgerMenu />);
        });


        it('matches snapshot', () => {
            const comp = renderer.create(<BurgerMenu />);
            const app = comp.toJSON();
            expect(app).toMatchSnapshot();
        });

        it('exists', () => {
            expect(wrapper.exists).toBeTruthy();
        });

        it('renders a <Toggle/> component if isOpen === false', () => {
            expect(wrapper.find(Toggle).length).toBe(1);
            expect(wrapper.find(Navigation).length).toBe(0);
        });
    });


    it('renders a <Navigation/> component if isOpen === true', () => {
        useShowHook.mockImplementation(() => ({
            isShow: true,
            reverseBoolean: jest.fn()
        }));
        const wrapper2 = shallow(<BurgerMenu />);
        expect(wrapper2.find(Toggle).length).toBe(1);
        expect(wrapper2.find(Navigation).length).toBe(1);
    });
});