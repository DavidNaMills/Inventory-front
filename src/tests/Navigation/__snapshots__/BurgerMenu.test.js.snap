// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BurgerMenu test auite with isShow set to false matches snapshot 1`] = `
<button
  className="hamburger hamburger--collapse"
  onClick={[Function]}
  style={Object {}}
  type="button"
>
  <span
    className="hamburger-box"
  >
    <span
      className="hamburger-inner"
    />
  </span>
</button>
`;
